pitch.md

```diff
@@ -23,38 +86,122 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
 by describing which unicorn properties (horn color, tattoo, etc.) are most 
 important to users and what kinds of values those properties might have._ 
 
+The user creates instruments by combining components. There are three types of
+components, distinguished by their inputs and outputs.
+
+A _source_ is a component which produces a signal. (A signal is like a
+potential sound. At runtime, it can be synthesized into a sound via the
+computer's DAC, or perhaps it can be written to a file.) A source has no input
+(though it may have _parameters_, see below) and a single output. There will be
+several types of primitive sources, distinguished by the characteristics of the
+signals they produce. Examples include frequency modulation synthesizers and
+physical modeling synthesizers.
+
+An _effect_ is a component which modifies a signal. Its input is a signal, and
+its output is a new, transformed signal. Examples of primitive effects may
+include equalization, compression, and distortion.
```

Comment:
Are effects limited by what is supported in CSound itself? I'm not too familiar with this domain, but are there certain effects that take precedence or are more important than others that you'll prioritize?